Gordon360/Services/ScheduleService.cs

```diff
@@ -24,18 +24,25 @@ public ScheduleService(CCTContext context)
         /// Fetch the schedule item whose id and session code is specified by the parameter
         /// </summary>
         /// <param name="username">The AD Username of the student</param>
+        /// <param name="sessionID">The selected session of the student</param>
         /// <returns>StudentScheduleViewModel if found, null if not found</returns>
-        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username)
+        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username, string? sessionID)
```

Comment:
If you make give `sessionID` a default value of `null`, you can just not pass the second argument when calling it with no `sessionID`:
```suggestion
        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username, string? sessionID = null)
```

For example, this
```csharp
var result = _scheduleService.GetScheduleStudentAsync(authenticatedUserUsername, null);
```
could become
```csharp
var result = _scheduleService.GetScheduleStudentAsync(authenticatedUserUsername);
```
which is (at least to me) much more sensible.

Reply:
It looks like you updated the interface but not the actual implementation of the method. Please make sure to update the method implementation as well.